android/src/test/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/HomeViewModelTest.kt

```diff
@@ -0,0 +1,80 @@
+package co.nimblehq.ic.kmm.suv.android.ui.screens.home
+
+import androidx.activity.compose.setContent
+import co.nimblehq.ic.kmm.suv.android.rule.MainCoroutinesRule
+import co.nimblehq.ic.kmm.suv.android.ui.screens.login.LoginViewModel
+import co.nimblehq.ic.kmm.suv.domain.model.AppError
+import co.nimblehq.ic.kmm.suv.domain.usecase.GetProfileUseCase
+import co.nimblehq.ic.kmm.suv.domain.usecase.LogInUseCase
+import co.nimblehq.ic.kmm.suv.helper.date.DateTime
+import co.nimblehq.ic.kmm.suv.helper.date.DateTimeFormatter
+import co.nimblehq.ic.kmm.suv.helper.date.DateTimeFormatterImpl
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import kotlinx.datetime.LocalDate
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class HomeViewModelTest {
+
+    private lateinit var viewModel: HomeViewModel
+
+    private val mockGetProfileUseCase: GetProfileUseCase = mockk()
+    private val mockDateTime: DateTime = mockk()
+
+    @ExperimentalCoroutinesApi
+    @get:Rule
+    val mainCoroutinesRule = MainCoroutinesRule()
+
+    @Before
+    fun setup() {
+        every { mockDateTime.today() } returns LocalDate(2022, 11, 8)
+        every { mockGetProfileUseCase() } returns flowOf(mockk())
+        viewModel = HomeViewModel(mockGetProfileUseCase, mockDateTime, DateTimeFormatterImpl())
+    }
+
+    @Test
+    fun `When its init, today should be right`() {
+        viewModel.currentDate.value shouldBe "TUESDAY, NOVEMBER 8"
+    }
+
+    @Test
+    fun `When load profile successfully, the avatar url should not be null`() = runTest {
+        viewModel.loadProfile()
+        advanceUntilIdle()
+
+        viewModel.avatarUrlString.value shouldNotBe null
```

Comment:
should we expect an exact avatar instead of only checking null?

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/100/commits/5ad895b5ebe7f785e4dbbc060325c0509b5b89a7